packages/hash/shared/src/save.ts

```diff
@@ -312,11 +316,61 @@ const updateBlocks = defineOperation(
                 },
               });
             }
+          } else {
+            const draftBlock = getDraftEntityFromEntityId(
+              entityStore.draft,
+              savedEntity.entityId,
+            );
+            const draftChildEntityId = draftBlock?.properties.entity?.entityId;
+            const draftChildEntityWithDraftId = getDraftEntityFromEntityId(
+              entityStore.draft,
+              draftChildEntityId,
+            );
+
+            if (draftChildEntityWithDraftId && draftChildEntityId) {
+              const { draftId: _, ...draftChildEntity } =
```

Comment:
removed `draftId` since it can affect the comparison below. entities in draft store (`store.draft`) have a draftId but entities in the saved store (`store.saved`) don't